### PR TITLE
Export `grid.draw()` methods allowing use of `ggsave()`

### LINKED ISSUE #107: Further instruction for saving files
Because a ggsurvfit figure needs to be built to add the risktable, users need to run `ggsurvfit_build()` before saving with `ggsave()`.

I need to either 1. add instructions for saving on the website, or add a function to save the figures.

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
* Added S3 methods `grid.draw.ggsurvfit()` and `grid.draw.ggcuminc()` which in turn allows us to save images from the package directly with `ggplot2::ggsave()` (#107)

**If there is an GitHub issue associated with this pull request, please provide link.**
closes #107

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [x] Ensure all package dependencies are installed by running `renv::install()`
- [x] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [x] If a new function was added, function included in `_pkgdown.yml`
- [x] If a bug was fixed, a unit test was added for the bug check
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Overall code coverage remains >99.5%. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Begin in a fresh R session without any packages loaded. 
- [x] R CMD Check runs without errors, warnings, and notes
- [x] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# ggsurvfit (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,18 +120,19 @@ The package also includes gems for those using the [CDISC ADaM
 ADTTE](https://www.cdisc.org/standards/foundational/adam/adam-basic-data-structure-bds-time-event-tte-analyses-v1-0)
 data model.
 
-If columns `"PARAM"` or `"PARAMCD"` are present in the data frame, their
-values will be used to construct default labels in the `ggsurvfit()`
-figure.
+If columns `"PARAM"` or `"PARAMCD"` are present in the data frame passed
```
Comment (SHAESEN2) [inline README.md:123]: ... and ... are present based on your code. It is also requirement of CDISC that both are present.

File: R/grid.draw_ggsurvfit.R [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#' Draw ggsurvfit object
+#'
+#' `grid::grid.draw()` methods for objects of classes 'ggsurvfit' and 'ggcuminc'.
+#' These are implemented to allow users to directly call `ggplot2::ggsave()`
```
Comment (SHAESEN2) [inline R/grid.draw_ggsurvfit.R:4]: Could we add the reason why they could not use it without grid.draw?

Comment (ddsjoberg) [reply R/grid.draw_ggsurvfit.R:4]: if only I understood all the details! 😆

Comment (SHAESEN2) [reply R/grid.draw_ggsurvfit.R:4]: Could we at least write high-level why we need a function to do this and why we could not use ggsave without drawing the object.

File: README.md (hunk 1 above)
Comment (SHAESEN2) [reply README.md:123]: I mean that both PARAM and PARAMCD are required to be present in the ADTTE as per CDISC. You will only use PARAM.

File: R/grid.draw_ggsurvfit.R (hunk 2 above)
Comment (ddsjoberg) [reply R/grid.draw_ggsurvfit.R:4]: I really do not know the details

File: README.md (hunk 1 above)
Comment (ddsjoberg) [reply README.md:123]: This is just a high-level overview from the README, and doesn't match 100% the requirements for the label to be used---and I am ok with that.

**Comment by bailliem** [comment]
Might need to update the renv snapshot. My windows install is throwing an error when building the site as `ggeasy` package is missing. It dosen't look like the package is  part of the renv lock file. 

```
-- RMarkdown error -------------------------------------------------------------
Quitting from lines 202-211 (gallery.Rmd) 
Error in loadNamespace(x) : there is no package called 'ggeasy'
In addition: Warning messages:
1: Tried to calculate with group_by(), but the calculation failed.
Falling back to ungrouped filter operation... 
2: Using `across()` in `filter()` is deprecated, use `if_any()` or `if_all()`. 
--------------------------------------------------------------------------------
Error in render_rmarkdown(pkg, input = input, output = output_file, output_format = format,  : 
  Failed to render RMarkdown
Caused by error:
! callr subprocess failed: there is no package called 'ggeasy'
Caused by error:
! there is no package called 'ggeasy'
Error: callr subprocess failed: Failed to render RMarkdown
Caused by error:
! callr subprocess failed: there is no package called 'ggeasy'
Caused by error:
! there is no package called 'ggeasy'
Type .Last.error.trace to see where the error occurred
```

**Review by bailliem — APPROVED** [review]
Lookjs good. My only comment , suggestion, is to add an example of saving in the gallery to reiterate this option. Maybe an example using good settings for a pubcliation ready image. For example, at the end there could be an example of saving say the KMcommunicate plot e.g. 

```
survfit2(Surv(time, status) ~ surg, data = df_colon) %>%
  ggsurvfit(linetype_aes = TRUE) +
  add_confidence_interval() +
  add_risktable(
    risktable_stats = c("n.risk", "cum.censor", "cum.event")
  ) +
  theme_ggsurvfit_KMunicate() +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(expand = c(0.02, 0)) +
  ggtitle("My published plot") +
  theme(legend.position = c(0.85, 0.85))

ggplot2::ggsave("publication-ready.png", dpi = 600, width = 12)
```

**Comment by ddsjoberg** [comment]
Hey @bailliem ! Thank you for the review!

- I added a section to the gallery about saving the images.
- Regarding the ggeasy pkg: The gallery is an article, rather than a vignette. The distinction is that an article is not shipped with the package. In the gallery article we use ggeasy and gghighlight. Because it's not a vignette, these are not proper dependencies of the package (which is great!). I am not sure if there is a way to tell R that these should be installed at any point, though. We're not using renv here to track dependencies.

**Pull request merged**